stores/metadata.go

```diff
@@ -761,8 +782,35 @@ func (s *SQLStore) SetContractSet(ctx context.Context, name string, contractIds
 			return err
 		}
 
+		// invalidate the health on all slab which are affected by this change
+		currentMap := make(map[fileContractID]struct{})
+		for _, fcid := range dbCurrentContracts {
+			currentMap[fcid] = struct{}{}
+		}
+		newMap := make(map[fileContractID]struct{})
+		for _, c := range dbNewContracts {
+			newMap[c.FCID] = struct{}{}
+		}
+		for _, c := range dbNewContracts {
+			delete(currentMap, c.FCID)
+		}
+		for _, fcid := range dbCurrentContracts {
+			delete(newMap, fcid)
+		}
+		diff := make([]fileContractID, 0, len(currentMap)+len(newMap))
+		for fcid := range currentMap {
+			diff = append(diff, fcid)
+		}
+		for fcid := range newMap {
+			diff = append(diff, fcid)
+		}
+		err = invalidateSlabHealthByFCID(tx, diff)
```

Comment:
This is fine but it could be a gorm hook on the join table entity? Might be neat to try that.